ansible/roles/layers-service/tasks/main.yml

```diff
@@ -0,0 +1,93 @@
+- include: ../../common/tasks/setfacts.yml
+
+- name: install Postgis and GDAL Execs
+  apt: pkg={{ item }} state=present update_cache=yes
+  with_items:
+    - postgis
+    - gdal-bin
+
+- name: copy transient files to tmp (schemas etc)
+  copy: src={{item}} dest=/tmp
+  with_items:
+    - layersdb.sql
+    - stored_procs.sql    
+  tags:
+    - createdb
+
+# Not working currently.....
+- name: setup the DB - WARNING!!! this isnt idempotent yet
+  command: "psql -d {{ layers_db_name | default('layersdb') }} -a -f /tmp/layersdb.sql"
+  sudo_user: postgres
+  sudo: True  
+  tags:
+    - createdb
+
+- name: setup stored procedures - WARNING!!! this isnt idempotent yet
+  command: "psql -d {{ layers_db_name | default('layersdb') }} -a -f /tmp/stored_procs.sql"
+  sudo_user: postgres
+  sudo: True  
+  tags:
+    - createdb
+
+- name: setup the DB user password
```

Comment:
...you can then add a 'when' condition to this task so the script is only executed if the db was just created:
```
when: dbschema.changed
```